src/home/event_reaction_list.rs

```diff
@@ -0,0 +1,326 @@
+                        if id.0 != *index {
+                            return;
+                        }
+                        if !widget_ref.area().rect(cx).contains(e.abs) {
+                            if self.event_reaction_list.get(id.0 as usize).is_some() {
+                                reset_tooltip_state = true;
+                                cx.widget_action(uid, &scope.path, RoomScreenTooltipActions::HoverOut);
+                            }
+                        }
+                    });
+                    // If the mouse does not leave this particular reaction button, propagate tooltip actions.
+                    if !reset_tooltip_state {
+                        cx.widget_action(uid, &scope.path, tooltip_actions.clone());
+                    }
+                }
+                if reset_tooltip_state {
+                    self.tooltip_state = None;
+                }
+            } 
+        }
+        if let Event::Actions(actions) = event {
+            self.children
+            .iter()
+            .for_each(|(_id, widget_ref)| {
+                if widget_ref.clicked(actions) {
+                    let text = widget_ref.text();
+                    let reaction_string = text.rsplit_once(' ')
+                    .map(|(prefix, _)| prefix)
+                    .unwrap_or(&text);
+                    if let Some(key) = emojis::get_by_shortcode(reaction_string) {
+                        submit_async_request(MatrixRequest::ToggleReaction {
+                            room_id: room_id.clone(),
+                            timeline_event_id: timeline_event_id.clone(),
+                            reaction: key.as_str().to_string(),
+                        });
+                    }
+                }
+            });
+        }
+        
+        
+    }    
+}
+
+impl ReactionListRef {
+    /// Set the list of reactions and their counts to display in the ReactionList widget,
+    /// along with the room ID and event ID that these reactions are for.
+    ///
+    /// This will clear any existing list of reactions and replace it with the given one.
+    ///
+    /// The given `event_tl_item_reactions` is a map from each reaction's raw string (including any variant selectors)
+    /// to the list of users who have reacted with that reaction.
+    ///
+    /// The given `room_id` is the ID of the room that these reactions are for.
+    ///
+    /// The given `timeline_event_item_id` is the ID of the event that these reactions are for.
+    /// Required by Matrix API
+    pub fn set_list(
+        &mut self,
+        cx: &mut Cx,
+        event_tl_item_reactions: &ReactionsByKeyBySender,
+        room_id: OwnedRoomId,
+        timeline_event_item_id: TimelineEventItemId,
+    ) {
+        let Some(client_user_id) = get_client().and_then(|c| c.user_id().map(|user_id| user_id.to_owned()) ) else { return };
+        if let Some(mut instance) = self.borrow_mut() {
```

Comment:
nit: use an early-return flow to reduce indentation of the rest of the function block and improve readability:
```suggestion
        let Some(mut instance) = self.borrow_mut() else { return };
```